src/dev-server/utils.ts

```diff
@@ -0,0 +1,141 @@
+
+        this.push(resultData);
+        this.includePrefix = includeSuffix;
+
+        callback();
+    }
+}
+
+export const pipeLogsWithPrefix = (childProcess: ChildProcessWithoutNullStreams, prefix: string): void => {
+    const logOnErrorCb = (error: Error | null): void => {
+        if (error) {
+            logger.error("Got an error trying to pipeline dev server logs:", error.message);
+        }
+    };
+
+    pipeline(childProcess.stdout, new WithPrefixTransformer(prefix), process.stdout, logOnErrorCb);
+    pipeline(childProcess.stderr, new WithPrefixTransformer(prefix), process.stderr, logOnErrorCb);
+};
+
+const defaultIsReadyFn = (response: Awaited<ReturnType<typeof globalThis.fetch>>): boolean => {
+    return response.status >= 200 && response.status < 300;
+};
+
+export const waitDevServerReady = async (
+    devServer: ChildProcessWithoutNullStreams,
+    readinessProbe: Config["devServer"]["readinessProbe"],
+): Promise<void> => {
+    if (typeof readinessProbe !== "function" && !readinessProbe.url) {
+        return;
+    }
+
+    logger.log("Waiting for dev server to be ready");
+
+    if (typeof readinessProbe === "function") {
+        return Promise.resolve()
+            .then(() => readinessProbe(devServer))
+            .then(() => logger.log("Dev server is ready"));
+    }
+
+    const isReadyFn = readinessProbe.isReady || defaultIsReadyFn;
+
+    let isSuccess = false;
+    let isError = false;
+
+    const timeoutPromise = new Promise<never>((_, reject) => {
+        setTimeout(() => {
+            if (!isError && !isSuccess) {
+                isError = true;
+                reject(new Error(`Dev server is still not ready after ${readinessProbe.timeouts.waitServerTimeout}ms`));
+            }
+        }, readinessProbe.timeouts.waitServerTimeout).unref();
+    });
+
+    const readyPromise = new Promise<void>(resolve => {
+        const tryToFetch = async (): Promise<void> => {
+            const signal = AbortSignal.timeout(readinessProbe.timeouts.probeRequestTimeout);
+
+            try {
+                const response = await fetch(readinessProbe.url!, { signal });
+                const isReady = await isReadyFn(response);
+
+                if (!isReady) {
+                    throw new Error("Dev server is not ready yet");
+                }
+
+                if (!isError && !isSuccess) {
+                    isSuccess = true;
+                    logger.log("Dev server is ready");
+                    resolve();
+                }
+            } catch (error) {
+                const err = error as { cause?: { code?: string } };
+
+                if (!isError && !isSuccess) {
+                    setTimeout(tryToFetch, readinessProbe.timeouts.probeRequestInterval).unref();
+
+                    const errorMessage = err && err.cause && (err.cause.code || err.cause);
+
+                    if (errorMessage && errorMessage !== "ECONNREFUSED") {
```

Comment:
`ECONNREFUSED` is thrown if server is still not ready.
If it is no `ECONNREFUSED`, it could be usefull